### PR TITLE
feat(typescript-estree): align nodes with estree 2020

### LINKED ISSUE #1950: Conform dynamic imports `import('a')` to the new ESTree spec
For this code example:
```ts
const x = import('module');
```

We currently emit the following AST (excluding the variable decl):
```json
{
  "type": "CallExpression",
  "callee": {
    "type": "Import",
  },
  "arguments": [
    {
      "type": "Literal",
      "raw": "'a'",
      "value": "a",
    }
  ],
  "optional": false,
}
```

When we should be emitting this (as per the [ESTree spec](https://github.com/estree/estree/blob/773c7358205ad7ab3963c8973f1b42ade36e6166/es2020.md#importexpression))
```json
{
  "type": "ImportExpression",
  "source": {
    "type": "Literal",
    "value": "a",
    "raw": "'a'"
  }
}
```

### LINKED ISSUE #1999: Conform bigint literals to the new ESTree spec
For this code example:
```ts
const x = 1n;
```

We currently emit the following AST (excluding the variable decl):
```json
{
  "type": "BigIntLiteral",
  "raw": "1n",
  "value": "1",
}
```

When we should be emitting this (as per the [ESTree spec](https://github.com/estree/estree/blob/773c7358205ad7ab3963c8973f1b42ade36e6166/es2020.md#bigintliteral)
```json
{
  "type": "Literal",
  "raw": "1n",
  "value": 1n,
  "bigint": "1"
}
```

### PR DESCRIPTION
**Pull request by armano2**
BREAKING CHANGE:

This PR aims to align new nodes with es2020 spec, mainly `BigIntLiteral` becomes `Literal` type with additional field `bigint` and `ImportExpression` is used instead of `CallExpression`.

https://github.com/estree/estree/blob/master/es2020.md

-----------------

### TODO/ISSUES:
- [x] update `Bigint` implementation
- [x] update `ImportExpression` implementation
- [x] update snapshots generation to support bigint
- [x] drop node 8

#### dropped:
- ~Jest does no support bigint in snapshots - solved by jest@25 (next)~
- ~`ts-jest` is not officially supporting jest@25 - https://github.com/kulshekhar/ts-jest/pull/1232~

Fixes #1999
Fixes #1950

### CODE REVIEW

File: packages/typescript-estree/tests/lib/__snapshots__/javascript.ts.snap [hunk 1]
```diff
@@ -132859,7 +132863,7 @@ Object {
               "pattern": "[\\\\u{0000000000000061}-\\\\u{7A}]",
             },
             "type": "Literal",
-            "value": Object {},
+            "value": /\\[\\\\u\\{0000000000000061\\}-\\\\u\\{7A\\}\\]/u,
```
Comment (armano2) [inline packages/typescript-estree/tests/lib/__snapshots__/javascript.ts.snap]: regexp will now be correctly stored in snapshots

File: packages/typescript-estree/tests/lib/__snapshots__/javascript.ts.snap [hunk 2]
```diff
@@ -16506,8 +16507,8 @@ Object {
           4,
         ],
         "raw": "0b1n",
-        "type": "BigIntLiteral",
-        "value": "0b1",
+        "type": "Literal",
+        "value": 1n,
```
Comment (armano2) [inline packages/typescript-estree/tests/lib/__snapshots__/javascript.ts.snap]: in node 8 this value is null, this will cause errors in snapshots

**Comment by fisker** [comment]
Can't parse BigInt `200_200n`

**Comment by bradzacher** [comment]
please don't comment on closed PRs - file a new issue instead, as per our contributing guidelines

https://github.com/typescript-eslint/typescript-eslint/blob/master/CONTRIBUTING.md#raising-issues

**Pull request merged**